components/builder-minio/habitat/hooks/post-run

```diff
@@ -1,19 +1,15 @@
-#!/bin/sh
+#!{{pkgPathFor "core/bash"}}/bin/bash
 
 exec 2>&1
 
-export AWS_ACCESS_KEY_ID="{{cfg.key_id}}"
-export AWS_SECRET_ACCESS_KEY="{{cfg.secret_key}}"
+source {{ pkg.svc_config_path }}/hook-helper.sh
 
-{{ #if cfg.use_ssl }}
-MINIO_ENDPOINT="--endpoint-url https://localhost:9000 --no-verify-ssl"
-{{else ~}}
-MINIO_ENDPOINT="--endpoint-url http://localhost:9000"
-{{ /if }}
+# Give service some time to start
+sleep $((RANDOM % 20 + 10))
```

Comment:
Do we need the random sleep for standalone mode?  We're going to hit this in a lot of cases.

Reply:
I agree with you that this part is bad. I'm also don't like random waits. I did it because of habitat bug: post-run hook will not re-run on failure (https://github.com/habitat-sh/habitat/issues/2364). It is hard to guess how much time minio will spend at first start so just wait. I added delay. Because there are might be number of nodes (in cluster) I added splay (random).
If bug with post-run will be resolved I believe this code must be deleted, but better to leave a tiny splay for cluster mode.

Also good point for standalone mode. I'll better change delay to 2 sec for standalone mode - more than enough. Will send update soon

Reply:
Cool, that works - thanks!